### ISSUE TITLE
Use the alias namespace and name for stylesheet tokens bundled with `@lwc/rollup-plugin`

### ISSUE THREAD

**Issue by jmsjtu** [salesforce/lwc#3455]
### Description
When an [aliased component](https://github.com/salesforce/lwc/blob/master/packages/%40lwc/module-resolver/README.md#alias-module-record) is bundled using `@lwc/rollup-plugin` the `stylesheetToken` contains the namespace and name of the original component rather than using the alias.

This is only an issue for components bundled using the `@lwc/rollup-plugin`.

#### Steps to Reproduce

Here's a [playground example](https://stackblitz.com/edit/salesforce-lwc-6whvsk?file=src%2Fmodules%2Fx%2Ffoo%2Ffoo.js,src%2Fmodules%2Fx%2Ffoo%2Ffoo.html,src%2Fmodules%2Fx%2Fapp%2Fapp.html,lwc.config.json,src%2Fmodules%2Fx%2Fapp%2Fapp.js).

```javascript
{
    "modules": [
        {
            "dir": "src/modules"
        },
        {
            "name": "c/baz",
            "path": "src/modules/x/foo/foo.js"
        }
    ]
}
// lwc.config.json
```

```html
<template>
    <h1>Hello world!</h1>
    <c-baz></c-baz>
</template>
```

The `stylesheetToken` for `c-baz` renders as:

<img width="745" alt="Screenshot 2023-04-10 at 12 56 17 PM" src="https://user-images.githubusercontent.com/8368171/231003304-5e289266-625a-4e8e-ba30-148b2a1a0c30.png">
```

#### Expected Results
The `stylesheetToken` should use the aliased namespace and name, in the above example `c-baz_baz`. 

#### Actual Results
The `stylesheetToken` uses the original namespace and name in the token, `x-foo_foo` in the above example.


### Version
<!--
Node project
package.json
versions < 0.17: "lwc-engine": "0.16.8"
versions >= 0.17: "lwc-engine": "0.17.2"
-->
- LWC: 2.41.1


**Possible Solution**
In the `@lwc/rollup-plugin` have HTML imports inherit their namespace and name from the component that imports them.

In the above example, since `x/foo` is aliased as `c/baz`, the implicit HTML import should have the `c/baz` namespace/name.

**Comment by nolanlawson** [comment]
Could this also be solved by #3313 ? (Specifically, by randomizing the scope token into something like `lwc-<hash>`.)

**Comment by jmsjtu** [comment]
Yeah I think that would be a great idea!

The only real issue I saw with this was that it could be confusing to component authors to see a tag with different `stylesheetToken` assigned than the tag name.

 For example:

```html
<c-baz class="x-foo_foo"></c-baz>
```

I think if we can replace the name with a hash like `lwc-<hash>` there would be much less confusion.

**Comment by jmsjtu** [comment]
Resolved with #3580
